### PR TITLE
Initial SOMADataFrame class

### PR DESCRIPTION
**Pull request by aaronwolen**
Initial `SOMADataFrame` class with basic functionality for creation, writing, and reading. 

Other changes:
- Removed outdated tests from the original package

### CODE REVIEW

File: apis/r/R/SOMADataFrame.R [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+      # soma_rowid should not be included in the results
+      tiledb::extended(arr) <- FALSE
+
+      # select columns
+      tiledb::attrs(arr) <- column_names %||% character()
+
+      # select ranges
+      if (!is.null(ids)) {
+        tiledb::selected_ranges(arr) <- list(soma_rowid = cbind(ids, ids))
```
Comment (johnkerl) [inline apis/r/R/SOMADataFrame.R:139]: does this 'stick' in place after the `read` has returned?

File: apis/r/R/SOMADataFrame.R [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      }
+
+      # filter
+      if (!is.null(value_filter)) {
+        stopifnot(is_scalar_character(value_filter))
+        tiledb::query_condition(arr) <- do.call(
+          what = tiledb::parse_query_condition,
+          args = list(expr = str2lang(value_filter), ta = self$object)
```
Comment (johnkerl) [inline apis/r/R/SOMADataFrame.R:147]: TIL `str2lang`! :D

File: apis/r/R/SOMADataFrame.R (hunk 1 above)
Comment (johnkerl) [reply apis/r/R/SOMADataFrame.R:139]: bigger picture, `tiledb::selected_ranges` in the `tiledb-r` API is stateful while `read` in the `tiledb-soma` API is stateless

**Review by johnkerl — CHANGES_REQUESTED** [review]
this is awesome!! :) 

only issue to be resolved/discussed/something is statefulness of `tiledb::selected_ranges` being used here in a stateless SOMA query API

File: apis/r/R/SOMADataFrame.R (hunk 1 above)
Comment (aaronwolen) [reply apis/r/R/SOMADataFrame.R:139]: I make a copy of the array pointer (`arr <- self$object`) to keep it stateless, e.g.:

```r
library(tiledbsoma)
uri <- file.path(tempdir(), "soma-dataframe")
asch <- arrow::schema(
  foo = arrow::int32(),
  bar = arrow::float64(),
  baz = arrow::string()
)

sdf <- SOMADataFrame$new(uri)
sdf$create(asch)

tbl <- arrow::arrow_table(
  soma_rowid = 0L:9L, foo = 1L:10L, bar = 1.1:10.1, baz = letters[1:10]
)

sdf$read(value_filter = "foo < 5")
# Table
# 4 rows x 3 columns
# $foo <int32>
# $bar <double>
# $baz <string>

sdf$read()
# Table
# 10 rows x 3 columns
# $foo <int32>
# $bar <double>
# $baz <string>
```

Comment (johnkerl) [reply apis/r/R/SOMADataFrame.R:139]: nice, totally missed that!!

**Review by johnkerl — APPROVED** [review]

**Pull request merged**